portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/StructuredContentFilteringAndSorting.testcase

```diff
@@ -0,0 +1,142 @@
+@component-name = "portal-headless-frontend-infrastructure"
+definition {
+
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.main.component.name = "Headless Frontend Infrastructure";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+
+		task ("Given a content structure created") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "guest");
+
+			NavItem.gotoStructures();
+
+			WebContentStructures.addCP(structureName = "content-structure");
+
+			FormViewBuilder.addFieldByDoubleClick(fieldType = "Text");
+
+			FormFields.editFieldReference(key_fieldReference = "Text");
+
+			Button.clickSave();
+		}
+
+		task ("Given I create four structuredContent draft with different priority field set") {
+			var ddmStructureId = WebContentStructures.getDdmStructureId(structureName = "content-structure");
+
+			var responseToParse1 = HeadlessWebcontentAPI.createStructuredContentDraft(
+				data = "Text",
+				ddmStructureId = "${ddmStructureId}",
+				label = "Text",
+				name = "Text",
+				priority = "1.0",
+				title = "Web Content 1");
+			var responseToParse2 = HeadlessWebcontentAPI.createStructuredContentDraft(
+				data = "Text",
+				ddmStructureId = "${ddmStructureId}",
+				label = "Text",
+				name = "Text",
+				priority = "1.0",
+				title = "Web Content 2");
+			var responseToParse3 = HeadlessWebcontentAPI.createStructuredContentDraft(
+				data = "Text",
+				ddmStructureId = "${ddmStructureId}",
+				label = "Text",
+				name = "Text",
+				priority = "2.99",
+				title = "Web Content 3");
+			var responseToParse4 = HeadlessWebcontentAPI.createStructuredContentDraft(
+				data = "Text",
+				ddmStructureId = "${ddmStructureId}",
+				label = "Text",
+				name = "Text",
+				priority = "0.99",
+				title = "Web Content 4");
+		}
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			var response = HeadlessWebcontentAPI.returnAllstructuredContentId();
+
+			HeadlessWebcontentAPI.deleteAllStructuredContent(structuredContentId = "${response}");
+
+			WebContentStructures.tearDownCP();
+		}
+	}
+
+	@priority = "4"
+	test CanFilterEqualToPriorityValuesSortedByPriorityAscendingByDefault {
+		property portal.acceptance = "true";
+
+		task ("When web contents are filtered with 1.0 priority and sorted with default priority") {
+			var response = HeadlessWebcontentAPI.filterAndSortAdminStructuredContent(
+				filtervalue = "priority%20eq%201.0",
+				sortvalue = "priority");
+		}
+
+		task ("Then the response body only includes items with priority 1.0") {
+			HeadlessWebcontentAPI.assertTitleFieldWithCorrectName(
+				expectedTitleName = "Web Content 1,Web Content 2",
+				responseToParse = "${response}");
+		}
```

Comment:
When no sort is provided, the default sorting is by id (entryClassPK on ElasticSearch).

When multiple sorts are provided it will sort by the next property in case the previous property is equal.

When there is no other property to sort by, I've made some tests and I think it is using the modified date to sort the results with the same value. So in case the sort is ascending, it will sort the old entries first and then the new ones. In case the sort is descending, it will sort the new entries first and then the old ones.

I've got this conclusion by an empirical approach but seems consistent.

To get the results we rely on ElasticSearch and there are no more sort properties in the ElasticSearch query. Searching about how ElasticSearch orders the results, they say it is more or less random, and what defines the order is the score of each document. I guess we are considering the modified date field to improve the score of each document and that's why it seems to take this field into account when there are no more fields to order by